cocos/scene-graph/node.ts

```diff
@@ -1704,12 +1707,11 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
      * @zh 这个节点的空间变换信息在当前帧内是否有变过？
      */
     get hasChangedFlags () {
-        return this._flagChangeVersion === globalFlagChangeVersion ? this._hasChangedFlags : 0;
+        return (this._versionedChangedFlags >>> 3) === globalFlagChangeVersion ? (this._versionedChangedFlags & 7) : 0;
     }
 
     set hasChangedFlags (val: number) {
-        this._flagChangeVersion = globalFlagChangeVersion;
-        this._hasChangedFlags = val;
+        this._versionedChangedFlags = (globalFlagChangeVersion << 3) | val;
```

Comment:
It is better to use a function to wrap them.

Reply:
This code snippet is small enough, it is not necessary to encapsulate it into a method. I have added explanations for the field _versionedChangedFlags.